nengo/synapses.py

```diff
@@ -385,12 +468,12 @@ class NoD(Step):
         """
 
         def __call__(self, t, signal):
-            self.X[:] = np.dot(self.A, self.X) + self.B * signal
-            return np.dot(self.C, self.X)[0]
+            self.X[:] = np.dot(self.X, self.AT) + signal[..., None] * self.BT
+            return np.dot(self.X, self.CT)[..., 0]
```

Comment:
Would `.squeeze(axis=-1)` be more failsafe or robust?

Reply:
Yes, probably a good idea.

Reply:
I left `OneXScalar` untouched, just in case `squeeze` has a bit of extra overhead. I changed the rest, though.